test/integration/deployment/deployment_test.go

```diff
@@ -498,3 +499,111 @@ func TestRollbackDeploymentRSNoRevision(t *testing.T) {
+	rsSelector, err := metav1.LabelSelectorAsSelector(testers[0].deployment.Spec.Selector)
+	if err != nil {
+		t.Fatalf("failed to parse deployment selector: %v", err)
+	}
+	rsList, err := c.ExtensionsV1beta1().ReplicaSets(ns.Name).List(metav1.ListOptions{LabelSelector: rsSelector.String()})
+	if err != nil {
+		t.Fatalf("failed to list replicasets of deployments %q and %q in namespace %q: %v", firstDeploymentName, secondDeploymentName, ns.Name, err)
+	}
+	rss := rsList.Items
+	if len(rss) != 2 {
+		t.Fatalf("expected 2 replicasets to be found, but found %d", len(rss))
+	}
+
+	firstController := metav1.GetControllerOf(&rss[0])
+	if firstController == nil {
+		t.Fatalf("replicaset %q is not owned by any deployment", rss[0].Name)
+	}
+	secondController := metav1.GetControllerOf(&rss[1])
+	if secondController == nil {
+		t.Fatalf("replicaset %q is not owned by any deployment", rss[1].Name)
+	}
+
+	firstDeploymentUID := testers[0].deployment.UID
+	secondDeploymentUID := testers[1].deployment.UID
+	if !((firstController.UID == firstDeploymentUID && secondController.UID == secondDeploymentUID) ||
+		(firstController.UID == secondDeploymentUID && secondController.UID == firstDeploymentUID)) {
+		t.Fatalf("expected replicasets %q and %q are owned by deployments %q and %q, but found deployments %q and %q", rss[0].Name, rss[1].Name, firstDeploymentName, secondDeploymentName, firstController.Name, secondController.Name)
+	}
+
+	// After making sure both replicasets are owned by intended deployments, store [replicaset's UID:controllerRef's UID] mapping for future usage
+	mapping := map[types.UID]types.UID{rss[0].UID: firstController.UID, rss[1].UID: secondController.UID}
+
+	// Scale only the first deployment by 1
+	newReplicas := replicas + 1
+	testers[0].deployment, err = testers[0].updateDeployment(func(update *v1beta1.Deployment) {
+		update.Spec.Replicas = &newReplicas
+	})
+	if err != nil {
+		t.Fatalf("failed updating deployment %q: %v", firstDeploymentName, err)
+	}
+
+	// Wait for the controller to notice the scale
+	if err := testers[0].waitForObservedDeployment(testers[0].deployment.Generation); err != nil {
+		t.Fatalf("encountered an error when controller waits to notice the scale: %v", err)
+	}
+
+	// Verify that only replicaset of first deployment is scaled
+	rsList, err = c.ExtensionsV1beta1().ReplicaSets(ns.Name).List(metav1.ListOptions{LabelSelector: rsSelector.String()})
+	if err != nil {
+		t.Fatalf("failed to list replicasets of deployments %q and %q in namespace %q: %v", firstDeploymentName, secondDeploymentName, ns.Name, err)
+	}
+	rss = rsList.Items
+	if len(rss) != 2 {
+		t.Fatalf("expected 2 replicasets to be found, but found %d", len(rss))
+	}
+	if !((*rss[0].Spec.Replicas == newReplicas && mapping[rss[0].UID] == firstDeploymentUID && *rss[1].Spec.Replicas == replicas) ||
+		(*rss[0].Spec.Replicas == replicas && *rss[1].Spec.Replicas == newReplicas && mapping[rss[1].UID] == firstDeploymentUID)) {
```

Comment:
This can be simplified with the map, too.

Reply:
Do you mean something like this:
```
// Scale only the first deployment by 1
newReplicas := replicas + 1
testers[0].deployment, err = testers[0].updateDeployment(func(update *v1beta1.Deployment) {
	update.Spec.Replicas = &newReplicas
})
if err != nil {
	...
}

// Wait for the controller to notice the scale
if err := testers[0].waitForObservedDeployment(testers[0].deployment.Generation); err != nil {
	...
}

// Verify that only replicaset of first deployment is scaled
if rs, _ := mapping[testers[0].deployment.UID]; *rs.Spec.Replicas != newReplicas {
	t.Errorf("expected replicaset %q of deployment %q is scaled from %d to %d, but found %d replicas", rs.Name, firstDeploymentName, replicas, newReplicas, *rs.Spec.Replicas)
}

......
```